src/mam4xx/aer_rad_props.hpp

```diff
@@ -0,0 +1,499 @@
+  // trop_level(:)
+  // zi(:,:) !geopotential height above surface at interfaces [m]
+  // ext_cmip6_lw_inv_m(:,:,:) !long wave extinction in the units of [1/m]
+
+  //! intent-inouts
+  //  odap_aer(:,:,:)  ! [fraction] absorption optical depth, per layer
+  //  [unitless]
+
+  // !local
+  // integer :: icol, ilev_tropp
+  // real(r8) :: lyr_thk !layer thickness [m]
+  // do icol = 1, ncol
+  // ilev_tropp = trop_level(icol) !tropopause level
+  const Real lyr_thk =
+      zi(ilev_tropp) - zi(ilev_tropp + 1); //! compute layer thickness in meters
+  constexpr Real half = 0.5;
+  //! update taus with 50% contributuions from the volcanic input file
+  //! and 50% from the existing model computed values at the tropopause layer
+  for (int i = 0; i < nlwbands; ++i) {
+    odap_aer(ilev_tropp, i) =
+        half * (odap_aer(ilev_tropp, i) +
+                (lyr_thk * ext_cmip6_lw_inv_m(ilev_tropp, i)));
+  }
+
+} // compute_odap_volcanic_at_troplayer_lw
+
+KOKKOS_INLINE_FUNCTION
+void compute_odap_volcanic_above_troplayer_lw(const int ilev_tropp,
+                                              const ConstColumnView &zi,
+                                              const View2D &ext_cmip6_lw_inv_m,
+                                              const View2D &odap_aer) {
+
+  //     !Above the tropopause, the read in values from the file include both
+  //     the stratospheric
+  // !and volcanic aerosols. Therefore, we need to zero out odap_aer above the
+  // tropopause !and populate it exclusively from the read in values.
+
+  // !intent-ins
+  // integer, intent(in) :: pver, ncol
+  // integer, intent(in) :: trop_level(:)
+
+  // real(r8), intent(in) :: zi(:,:) !geopotential height above surface at
+  // interfaces [m] real(r8), intent(in) :: ext_cmip6_lw_inv_m(:,:,:) !long wave
+  // extinction in the units of [1/m]
+
+  // !intent-inouts
+  // real(r8), intent(inout) :: odap_aer(:,:,:) ! [fraction] absorption optical
+  // depth, per layer [unitless]
+
+  // !local
+  // integer :: ipver, icol, ilev_tropp
+  // real(r8) :: lyr_thk !layer thickness [m]
+
+  // !As it will be more efficient for FORTRAN to loop over levels and then
+  // columns, the following loops !are nested keeping that in mind
+
+  for (int kk = 0; kk < ilev_tropp; ++kk) {
+    const Real lyr_thk =
+        zi(kk) - zi(kk + 1); // ! compute layer thickness in meters
+    for (int i = 0; i < nlwbands; ++i) {
+      odap_aer(kk, i) = lyr_thk * ext_cmip6_lw_inv_m(kk, i);
+    } // end i
+
+  } // end kk
+
+} // compute_odap_volcanic_above_troplayer_lw
+
+/* Read the tropopause pressure in from a file containging a climatology. The
+  ! data is interpolated to the current dat of year and latitude.
+  !
+  ! NOTE: The data is read in during tropopause_init and stored in the module
+  ! variable trop */
+
+#if 0
+  KOKKOS_INLINE_FUNCTION
+void tropopause_climate(lchnk,ncol,pmid,pint,temp,zm,zi,    &  ! in
+             tropLev,tropP,tropT,tropZ)   {
+// TO be ported...
```

Comment:
For another PR?

Reply:
Yes, I hope @jaelynlitz can help with this port.